syncwatcher.go

```diff
@@ -459,6 +450,23 @@ func relativePath(path string, folderPath string) string {
 	return path
 }
 
+// Returns a function to test whether a file should be ignored
+// Ignorefile must contain the absolute path to ".stignore". The returned
+// function takes the path of the file to be tested relative to its folders root
+func createIgnoreTest(folderPath string) func(relPath string) bool {
+	ignores := ignore.New(false)
+	ignores.Load(filepath.Join(folderPath, ".stignore"))
+	return func(relPath string) bool {
+		if strings.HasPrefix(relPath, versionsFolder) { return true }
+		for _, ignorePrefix := range tempFilePrefixes {
+			if strings.HasPrefix(relPath, ignorePrefix) &&
```

Comment:
You have to get the filename first and use this for comparison (at least for the prefix check). Currently, you only ignore temp files in the folder root.

Your check matches files with a `relPath` of e.g. `.syncthing.test.txt.tmp`, but you also want to ignore files with a `relPath` of `dir/.syncthing.test.txt.tmp`.

`filepath.Base(relPath)` should give you the filename.